### PR TITLE
Replace deprecated Lua code

### PR DESCRIPTION
**Pull request by mattsc**
There are three deprecated Lua functions that still need to be replaced in 1.14: wesnoth.tovconfig(), wesnoth.get_variable() and wesnoth.set_variable().  tovconfig() is done and the deprecation message disappears.

get_variable() is mostly done, but I've hit a problem that I do not know how to solve.  It's the [definition of wml.variables in data/lua/core.lua](https://github.com/wesnoth/wesnoth/blob/master/data/lua/core.lua#L260) itself that causes the remaining deprecation message.  Obviously, this is the one instance where the old function is still needed, and I am not familiar enough with how this works at this time to know what to do.  Any advice?

I'll do set_variable() once I have figured this out.

### CODE REVIEW

**Comment by jyrkive** [comment]
@CelticMinstrel Can you offer input about what to do with the implementation of `wml.variables`?

**Comment by gfgtdf** [comment]
well just make a local copy (cloasure) of the function (wesnoth.get_variable) before it gets overwritten later.

**Comment by mattsc** [comment]
@gfgtdf Thanks, that worked, see the new commit. Let me know if there's something wrong about how I do that, otherwise I'll go ahead and do the changes for `wesnoth.set_variable()` also.  There's over 120 of those...

**Comment by mattsc** [comment]
Note that I just squashed a couple commits, so if you looked before right now, that last commit is different.

**Comment by CelticMinstrel** [comment]
My only complaint is the missing `v` in the first commit message. :P

**Comment by mattsc** [comment]
> My only complaint is the missing v in the first commit message. :P
Whoops!  I assume that there's a way to change a commit message in a local branch, and I am sure Google can tell me how to do that ...

**Comment by CelticMinstrel** [comment]
Interactive rebase can do it, then you'd need to force-push.

**Comment by mattsc** [comment]
> Interactive rebase can do it, then you'd need to force-push.

Okay, done.  Still working on wesnoth.set_variable().

**Comment by CelticMinstrel** [comment]
Kinda surprised that there's so few deprecated things left to replace though. I thought there'd be a whole lot more. There are quite a large number of deprecated Lua functions and tables (mostly listed at the bottom of either core.lua or helper.lua), are they really all unused now?

**Comment by mattsc** [comment]
Well, you did replace a lot of them already. Just as an example, all the helper.distance_between() instances are switched over already.  But I'll go through those lists again tomorrow just to be sure.

**Comment by mattsc** [comment]
Still working on it.  There are indeed quite a few uses of deprecated helper functions left.

**Comment by mattsc** [comment]
As far as I can tell, this is all now.  I'll wait for 24 hours or so for comments, and to let the checks complete, and then merge it.

**Comment by CelticMinstrel** [comment]
Only other thing I'd suggest is removing `wesnoth.require "helper"` or equivalent from any files that no longer use it.

**Comment by mattsc** [comment]
> Only other thing I'd suggest is removing wesnoth.require "helper" or equivalent from any files that no longer use it.

Good point.  I thought that I had done that, but that was in my campaign, not mainline.  :-P  There are actually quite a few of them.

Question: A whole bunch of Lua files include this:
`local H = wesnoth.require "helper"`
`local W = H.set_wml_action_metatable {}`
and then the only use of either library is a single (or at most a couple) `W.message` or similar. Is there are reason why I should not use `wesnoth.wml_actions.message` in that case and get rid of both of those library inclusions.

File: data/lua/core.lua [hunk 1]
```diff
@@ -236,11 +236,12 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 	--[========[Basic variable access]========]
 
 	-- Get all variables via wml.all_variables (read-only)
+	local get_all_vars_local = wesnoth.get_all_vars
 	setmetatable(wml, {
 		__metatable = "WML module",
 		__index = function(self, key)
 			if key == 'all_variables' then
-				return wesnoth.get_all_vars()
+				return get_all_vars_local()
```
Comment (GregoryLundberg) [inline data/lua/core.lua]: Seems spurious. All the other changes are setting variables. This one is the only which is not, and it doesn't really DO anything.

**Comment by mattsc** [comment]
That change prevents the definition of wml.all_variables in core.lua to produce a deprecation warning, in the same way how it was done for get_variable() and set_variable().

**Comment by mattsc** [comment]
Oh, right!  I'd forgotten about that.  Okay, so if it's a fixed message or something along those lines, I can use `wesnoth.wml_actions.message`.  If the command allows the user to pass a string or coordinates or something, I better not.

**Comment by mattsc** [comment]
> Only other thing I'd suggest is removing wesnoth.require "helper" or equivalent from any files that no longer use it.

Done.

If the checks pass, I will squash 525ee9e and 0661df5 onto the commits they belong to, but other than that I think this is finished?

**Comment by mattsc** [comment]
The checks all passed, so I did the rebase.  Now it's running the checks again, but since I didn't actually change anything the outcome should be the same.  If there are no more comments, I will merge this sometime tomorrow (Sunday).

**Pull request merged**